src/main/java/teammates/ui/pagedata/AdminEmailLogPageData.java

```diff
@@ -168,21 +203,17 @@ private QueryParameters parseQuery(String query) throws ParseException, InvalidP
      * @return false if the logEntry fails the filtering process
      */
     public boolean shouldShowLog(EmailLogEntry logEntry) {
+        // Skip test data if the request is not showing all logs
+        boolean isShowTestData = !logEntry.isTestData() || shouldShowAll;
 
         if (q == null) {
             if (this.queryMessage == null) {
                 this.queryMessage = "Error parsing the query. QueryParameters not created.";
             }
-            return true;
+            return isShowTestData;
         }
 
         // filter based on what is in the query
-        if (q.isToDateInQuery && logEntry.getTime() > q.toDateValue) {
-            return false;
-        }
-        if (q.isFromDateInQuery && logEntry.getTime() < q.fromDateValue) {
```

Comment:
Are these lines not needed to hide the log entry if it does not fall within the user-specified `before:` `after:` range?